system/jlib/jtrace.hpp

```diff
@@ -55,6 +66,33 @@ class jlib_decl LogTrace
     void setLocalId(const char* id);
 };
 
+//class CHPCCHttpTextMapCarrier;
+
+interface ISpan : extends IInterface
+{
+    virtual void setSpanAttribute(const char * key, const char * val) = 0;
+    virtual void setSpanAttributes(const IProperties * attributes) = 0;
+    virtual void addSpanEvent(const char * eventName) = 0;
+
+    virtual void querySpanContextProperties(IProperties * contextProps) = 0;
+    //virtual bool injectSpanContext(CHPCCHttpTextMapCarrier * carrier) = 0;
+    virtual bool injectSpanContext(IProperties * contextProps) = 0;
```

Comment:
This name could be improved.  Probably get rather than inject.  What is the difference with the function above?  If different I suspect the parameter names should be different, and possibly combine into a single call.